apps/wakunode2/wakunode2.nim

```diff
@@ -513,10 +514,17 @@ proc startNode(node: WakuNode, conf: WakuNodeConf,
   if conf.peerExchangeNode != "":
     info "Retrieving peer info via peer exchange protocol"
     let desiredOutDegree = node.wakuRelay.parameters.d.uint64()
-    try:
-      discard await node.wakuPeerExchange.request(desiredOutDegree)
-    except:
-      return err("failed to retrieve peer info via peer exchange protocol: " & getCurrentExceptionMsg())
+    let pxPeersRes = await node.wakuPeerExchange.request(desiredOutDegree)
+    if pxPeersRes.isOk:
+      var record: enr.Record
+      var validPeers = 0
+      for pi in pxPeersRes.get().peerInfos:
+        if enr.fromBytes(record, pi.enr):
+          node.peerManager.addPeer(record.toRemotePeerInfo().get, WakuRelayCodec)
+          validPeers += 1
+      info "Retrieved peer info via peer exchange protocol", validPeers = validPeers
+    else:
+      warn "Failed to retrieve peer info via peer exchange protocol"
```

Comment:
This is a bit weird for me. While I agree with removing this from the `protocol` module, why not into `waku_node` when set up as a peer exchange client? It may make sense during node `init` to call some function that sets up peer exchange as a client and for now does only a single request. Basically we init and start all other discovery mechanisms when initiating the `WakuNode`, so this belongs there IMO.

Reply:
yes indeed, not sure what i was thinking. Addressed in https://github.com/waku-org/nwaku/pull/1527/commits/9083938d51fe615c47f08a37f12b1271832cd273

Added also a new test which also showcases how this is expected to be used.

https://github.com/waku-org/nwaku/blob/9083938d51fe615c47f08a37f12b1271832cd273/tests/v2/test_wakunode.nim#L293-L317